### DISCUSSION TITLE
Can Alpha Zero General handle games with imperfect information?

### DISCUSSION THREAD

**Discussion by Ross-Esmond** [suragnair/alpha-zero-general#310] [Q&A] [answered]
For example, games like poker where some information is unknown, or games where the next card to be drawn from a deck is unknown. A neural network shouldn't (generally) care if some information is unknown, but MCTS implementations may depend on determinism, especially if they need to run on the GPU.

**Comment by cestpasphoto** [comment, marked as answer]
The Alpha Zero algorithm (not only this implementation) is meant to work on perfect games. There are more recent algorithms like [ReBeL](https://ai.meta.com/blog/rebel-a-general-game-playing-ai-bot-that-excels-at-poker-and-more/) which are designed to support imperfect games such as poker.

However, by "torturing" a bit the Alpha Zero algorithm, I find it possible to support slightly random games (most information is predictable, only a small proportion is not - so not poker) and where same information is shared by all players. Btw it is also possible to make it support more than 2 players. But that is still kind of a hack.
